modular_skyrat/modules/reagent_forging/code/forge.dm

```diff
@@ -133,556 +173,523 @@
 
-		var/user_input = show_radial_menu(user, src, radial_options)
-		var/obj/item_to_spawn
+	if(!ceramic_item.forge_item)
+		fail_message(user, "cannot set [ceramic_item]")
+		return
 
-		if(!user_input)
-			fail_message(user, "No choice made")
-			return FALSE
+	balloon_alert_to_viewers("setting [ceramic_item]")
 
-		in_use = TRUE
-		balloon_alert_to_viewers("cooking...")
+	if(!do_after(user, ceramic_speed, target = src))
+		fail_message("stopped setting [ceramic_item]")
+		return
 
-		if(!do_after(user, 10 SECONDS, target = src))
-			fail_message(user, "You stop trying to cook [thing_to_cook]!")
-			in_use = FALSE
-			return FALSE
+	balloon_alert(user, "finished setting [ceramic_item]")
+	var/obj/item/ceramic/spawned_ceramic = new ceramic_item.forge_item(get_turf(src))
+	user.mind.adjust_experience(/datum/skill/production, 50)
+	spawned_ceramic.color = ceramic_item.color
+	qdel(ceramic_item)
+	in_use = FALSE
 
-		switch(user_input)
-			if("oven")
-				var/datum/component/bakeable/item_bakeable_component = thing_to_cook.GetComponent(/datum/component/bakeable)
-				item_to_spawn = item_bakeable_component.bake_result ? item_bakeable_component.bake_result : /obj/item/food/badrecipe
-			if("microwave")
-				item_to_spawn = thing_to_cook.microwaved_type ? thing_to_cook.microwaved_type : /obj/item/food/badrecipe
+/// Handles the creation of molten glass from glass sheets
+/obj/structure/reagent_forge/proc/handle_glass_sheet_melting(obj/attacking_item, mob/living/user)
+	in_use = TRUE
 
-		qdel(thing_to_cook)
-		new item_to_spawn(get_turf(src))
-		in_use = FALSE
-		return TRUE
+	if(forge_temperature < MIN_FORGE_TEMP)
+		fail_message(user, "forge too cool")
+		return
 
-	return ..()
+	var/obj/item/stack/sheet/glass/glass_item = attacking_item
+	var/glassblowing_speed = user.mind.get_skill_modifier(/datum/skill/production, SKILL_SPEED_MODIFIER) * BASELINE_ACTION_TIME
+	var/glassblowing_amount = BASELINE_HEATING_DURATION / user.mind.get_skill_modifier(/datum/skill/production, SKILL_SPEED_MODIFIER)
 
-/obj/structure/reagent_forge/proc/smelt_ore(obj/item/stack/ore/ore_item, mob/living/user)
-	var/skill_modifier = user.mind.get_skill_modifier(/datum/skill/smithing, SKILL_SPEED_MODIFIER)
+	balloon_alert_to_viewers("heating...")
 
-	if(in_use) //only insert one at a time
-		to_chat(user, span_warning("You cannot do multiple things at the same time!"))
+	if(!do_after(user, glassblowing_speed, target = src) || !glass_item.use(1))
+		fail_message(user, "stopped heating [glass_item]")
 		return
+
+	in_use = FALSE
+	var/obj/item/glassblowing/molten_glass/spawned_glass = new /obj/item/glassblowing/molten_glass(get_turf(src))
+	user.mind.adjust_experience(/datum/skill/production, 10)
+	COOLDOWN_START(spawned_glass, remaining_heat, glassblowing_amount)
+
+/// Handles creating molten glass from a metal cup filled with sand
+/obj/structure/reagent_forge/proc/handle_metal_cup_melting(obj/attacking_item, mob/living/user)
```

Comment:
This is done in a way I really don't like but can't do much about it until this pr is merged, as that'd cross into changing how the glassblowing system works.